src/Domain/Model/Resource.php

```diff
@@ -93,4 +97,50 @@ public function getAuthorId()
     {
       return $this->authorId;
     }
+
+    /**
+     * @return array
+     */
+    public function getVotes()
+    {
+        return $this->votes;
+    }
+
+    /**
+     * @param ResourceVote $vote
+     */
+    public function addVote(ResourceVote $vote)
+    {
+        $this->votes[] = $vote;
+        $this->votesTotal += $vote->getVote();
```

Comment:
I would change it to `$vote->getValue()` or something along those lines since it makes it more clear that there could be different values to the vote